web3.js/src/connection.ts

```diff
@@ -1755,10 +1837,20 @@ export class Connection {
     if (url.port !== null) {
       url.port = String(Number(url.port) + 1);
     }
-    this._rpcWebSocket = new RpcWebSocketClient(urlFormat(url), {
+
+    const websocketClientOptions: WebsocketClientOptions = {
       autoconnect: false,
       max_reconnects: Infinity,
-    });
+    };
+
+    if (httpHeaders) {
+      websocketClientOptions.headers = httpHeaders;
```

Comment:
I don't believe websocket clients support custom headers. Does this work in nodejs?

Reply:
> 
> 
> I don't believe websocket clients support custom headers. Does this work in nodejs?

Websocket clients should support it, adding the header to the initial HTTP request that establishes the websocket connection (which includes the Upgrade header etc). Some of the common nodejs libraries do support adding headers to the connect call.

Reply:
I double checked this is supported by the node client (https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketaddress-protocols-options).

Reply:
I believe WS APIs typically authenticate in the `subscribe` message rather than per-connection.  No harm in supporting both though, IMO.

Reply:
Cool, just to clarify, I meant that browser websocket clients don't support custom headers. Looks like this covers us in Node.js but as @t-nelson mentioned, we need a solution that will work in the browser as well

Reply:
True, the browser JS API doesn't have header support. For our use case for this we might just need to add a query string or possibly (mis)-use the websocket protocol header.

Reply:
Getting a bit off topic here, but seems like standards compliant way to do this is supply the bearer token as a query parameter - https://self-issued.info/docs/draft-ietf-oauth-v2-bearer.html#query-param.

Reply:
@jstarry oh, I got you backwards! I can pull the header portion from the websocket configuration since it wouldn't be consistent in both places.